File: Assets/Bundles/Shared/Resources/TS/Util/ArrayUtil.lua
-- Compiled with unity-ts v2.1.0-75
--[[
	*
	* Set of utilities for working with `Array` types.
]]
local ArrayUtil
do
	ArrayUtil = setmetatable({}, {
		__tostring = function()
			return "ArrayUtil"
		end,
	})
	ArrayUtil.__index = ArrayUtil
	function ArrayUtil.new(...)
		local self = setmetatable({}, ArrayUtil)
		return self:constructor(...) or self
	end
	function ArrayUtil:constructor()
	end
	function ArrayUtil:Combine(arrayA, arrayB)
		local _arrayB = arrayB
		local _arg0 = function(value)
			local _arrayA = arrayA
			local _value = value
			table.insert(_arrayA, _value)
		end
		for _k, _v in _arrayB do
			_arg0(_v, _k - 1, _arrayB)
		end
		return arrayA
	end
end
return {
	ArrayUtil = ArrayUtil,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Shared/Util/ArrayUtil.ts
/**
 * Set of utilities for working with `Array` types.
 */
export class ArrayUtil {
	/**
	 * Add arrayB to arrayA
	 * @param arrayA.
	 * @param arrayB.
	 * @returns A native Typescript array.
	 */
	public static Combine<T extends any[]>(arrayA: T, arrayB: T): T {
		arrayB.forEach((value) => {
			arrayA.push(value);
		});
		return arrayA;
	}
}
